pipeline/acquisition.py

```diff
@@ -0,0 +1,375 @@
+        cue_start_times = np.array(nwb['stimulus']['presentation']['cue_start']['timestamps'])
+        cue_end_times = np.array(nwb['stimulus']['presentation']['cue_end']['timestamps'])
+        pole_in_times = np.array(nwb['stimulus']['presentation']['pole_in']['timestamps'])
+        pole_out_times = np.array(nwb['stimulus']['presentation']['pole_out']['timestamps'])
+                
+        # form new key-values pair and insert key
+        key['n_trials'] = len(trial_names)
+        self.insert1(key)
+        print(f'Inserted trial set for session: Subject: {animal_id} - Date: {date_of_experiment}')
+        print('Inserting trial ID: ', end="")
+        
+        # loop through each trial and insert
+        for idx, trialId in enumerate(trial_names):
+            key['trial_id'] = trialId
+            # -- start/stop time
+            key['start_time'] = start_times[idx]
+            key['stop_time'] = stop_times[idx]
+            # -- events timing
+            key['cue_start_time'] = cue_start_times[idx]
+            key['cue_end_time'] = cue_end_times[idx]
+            key['pole_in_time'] = pole_in_times[idx]
+            key['pole_out_time'] = pole_out_times[idx]            
+            # form new key-values pair for trial_partkey and insert
+            self.Trial.insert1(key, ignore_extra_fields=True)
+            print(f'{trialId} ',end="")
+            # ======== Now add trial descriptors to the TrialInfo part table ====
+            # - good/bad trial_status (nwb['analysis']['good_trials'])
+            key['trial_status'] = 'good' if good_trials.flatten()[idx] == 1 else 'bad'
+            # - trial_type and trial_stim_status (nwb['epochs'][trial]['description']) 
+            key['trial_type'], key['trial_stim_status'] =  re.split(', ',trial_descs[idx])
+            # - trial_response (nwb['analysis']['trial_type_string'])
+            # note, the last type_string value is duplicated info of "stim"/"no stim" above, so ignore it here (hence the [idx,:-1])
+            match_idx = np.where(trial_type_mat[idx,:-1] == 1)
+            trial_response =  trial_type_string.flatten()[match_idx].item(0).decode('UTF-8')
+            if re.search('correct',trial_response.lower()) is not None:
+                trial_response = 'correct'
+            elif re.search('incorrect',trial_response.lower()) is not None:
+                trial_response = 'incorrect'
+            key['trial_response'] = trial_response.lower()
+            # insert
+            self.TrialInfo.insert1(key, ignore_extra_fields=True)
+        print('')
+    
+    
+@schema
+class BehaviorAcquisition(dj.Imported):
+    definition = """
+    -> Session
+    -> reference.BehavioralType
+    ---
+    behavior_time_stamp: longblob
+    behavior_timeseries: longblob        
+    """    
+      
+    
+@schema
+class ExtracellularAcquisition(dj.Imported):
+    definition = """
+    -> ExtracellularInfo
+    -> reference.ExtracellularType
```

Comment:
I don't quite understand why is `ExtracellularType` not already part of `ExtracellularInfo`